table/block_based_table_reader.cc

```diff
@@ -1436,23 +1504,20 @@ BlockBasedTable::BlockEntryIteratorState::NewSecondaryIterator(
   BlockHandle handle;
   Slice input = index_value;
   Status s = handle.DecodeFrom(&input);
-  auto iter = NewDataBlockIterator(table_->rep_, read_options_, handle, nullptr,
-                                   is_index_, s);
-  if (block_cache_cleaner_) {
-    uint64_t offset = handle.offset();
-    {
-      ReadLock rl(&cleaner_mu);
-      if (cleaner_set.find(offset) != cleaner_set.end()) {
-        // already have a reference to the block cache objects
-        return iter;
-      }
+  auto rep = table_->rep_;
+  if (block_map_) {
+    auto block = block_map_->find(handle.offset());
+    assert(block != block_map_->end());
+    if (LIKELY(block != block_map_->end())) {
```

Comment:
@maysamyabandeh please note in this code base, we don't do that. There are so many asserts in this code base and we don't try to work around when those asserts don't satisfy. I understand that your confident level of your new code is lower than existing code. That's why I suggest a fail fast solution. If your confident to the code quality is so low that you are concerning the fail fast happens so often that can cause trouble to users, I hope you add more tests to bring up your confident level, including adding it to the daily stress test coverage.

I suggest we invest on more robustness of code to handle hardware failures and user errors, rather than protecting from our own bugs.